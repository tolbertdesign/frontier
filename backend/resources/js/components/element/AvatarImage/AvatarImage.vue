<template>
  <img
    :alt="alt"
    :src="src"
    :class="[size, 'shadow border rounded-full']"
    @click="emit"
  >
</template>

<script>
export default {
  name: 'AvatarImage',
  status: 'prototype',
  release: '1.0.0',
  props: {
    alt: {
      type: String,
      default: '',
    },
    src: {
      type: String,
      default: '/v3-assets/dashboard/images/dashboard-avatar.svg',
    },
    shouldEmit: {
      type: Boolean,
      default: false,
    },
    small: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    size () {
      return this.small ? 'w-9 h-9' : 'w-16 h-16'
    },
  },
  methods: {
    emit () {
      if (this.shouldEmit) {
        this.$emit('open')
      }
    },
  },
}
</script>
