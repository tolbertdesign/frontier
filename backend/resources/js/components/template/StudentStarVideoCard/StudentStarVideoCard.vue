<template>
  <div class="flex w-full">
    <div class="mr-1 sm:mr-4">
      <a :href="editProfileLink">
        <AvatarImage
          :alt="participant.first_name"
          :src="avatarSrc"
        />
      </a>
    </div>
    <div class="flex-1 self-center">
      <h2>
        {{ participant.first_name }}
      </h2>
    </div>
    <div class="self-center text-right">
      <span
        v-if="inProgress"
        class="font-bold text-tertiary"
      >
        {{ lang.student_star_video.video_in_progress }}
      </span>
      <span
        v-else-if="readyToShare"
        class="font-bold text-primary"
      >
        {{ lang.student_star_video.ready_to_share }}
      </span>
      <RouterLink
        v-else
        :to="{ name: 'edit-participant', params: {id: participant.id} }"
        class="inline-block py-1 sm:py-2 px-4 sm:px-6 text-white hover:text-white shadow font-semibold rounded-full bg-secondary whitespace-no-wrap"
        @click.native="gaEvent('Parent Dashboard', 'Student Star Video Section', 'Upload Photo')"
      >
        {{ lang.student_star_video.upload_photo }}
      </RouterLink>
    </div>
  </div>
</template>

<script>
export default {
  name: 'StudentStarVideoCard',
  props: {
    participant: {
      type: Object,
      default: function () {
        return {}
      },
    },
  },
  computed: {
    avatarSrc () {
      return this.participant.profile.image_name == null ? '/v3-assets/dashboard/images/dashboard-avatar.svg' : this.avatarImgSrc
    },
    avatarImgSrc () {
      return this.$store.getters.avatarPath + this.participant.profile.image_name
    },
    editProfileLink () {
      return '/v3/tkdashboard/?redirect=' + encodeURI('auth/login/' + this.participant.fr_code + '/edit-personalize')
    },
    inProgress () {
      return this.participant.profile.image_name !== null && this.participant.profile.video_url == null
    },
    readyToShare () {
      return this.participant.profile.image_name !== null && this.participant.profile.video_url !== null
    },
    lang () {
      return this.$store.state.lang
    },
  },
}
</script>
