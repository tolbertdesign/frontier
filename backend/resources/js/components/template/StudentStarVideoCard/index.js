import StudentStarVideoCard from './StudentStarVideoCard'

export default StudentStarVideoCard
