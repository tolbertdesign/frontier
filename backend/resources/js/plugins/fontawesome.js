import Vue from 'vue'

import { library, dom } from '@fortawesome/fontawesome-svg-core'

// Brands (fab)
import { faFacebook, faFacebookF, faFacebookSquare, faTwitter } from '@fortawesome/free-brands-svg-icons'

// Regular (far)
import {
  faAddressCard,
  faArrowCircleRight,
  faBars as farBars,
  faCircle,
  faCommentAltLines,
  faEdit,
  faEnvelope,
  faExternalLinkSquare,
  faInfoCircle,
  faLock,
  faSignIn,
  faSignOut,
  faSquare,
  faTimes as farTimes,
  faUserEdit as farUserEdit,
} from '@fortawesome/pro-regular-svg-icons'

// Light (fal)
import { faBars, faTimes, faUserEdit } from '@fortawesome/pro-light-svg-icons'

// Solid Style Icons (fas)
import {
  faBars as fasBars,
  faBell,
  faBold,
  faCaretDown,
  faCaretUp,
  faCheck,
  faCheckSquare,
  faChevronDown,
  faChevronUp,
  faChevronLeft,
  faChevronRight,
  faCircle as fasCircle,
  faCode,
  faCoffee,
  faCog,
  faComment,
  faEdit as fasEdit,
  faEnvelope as fasEnvelope,
  faExclamationCircle,
  faExclamationTriangle,
  faExpandArrowsAlt,
  faEye,
  faItalic,
  faLink,
  faListOl,
  faListUl,
  faLock as fasLock,
  faMinus,
  faMobileAlt,
  faParagraph,
  faPlayCircle,
  faPlus,
  faQuestionCircle,
  faQuoteRight,
  faRedo,
  faShare,
  faShareSquare,
  faSortDown,
  faSpinner,
  faStrikethrough,
  faTimes as fasTimes,
  faTrashAlt,
  faUnderline,
  faUndo,
  faUser,
  faUsers,
  faUserPlus,
} from '@fortawesome/free-solid-svg-icons'

import {
  FontAwesomeIcon,
  FontAwesomeLayers,
  FontAwesomeLayersText,
} from '@fortawesome/vue-fontawesome'

library.add(
  faAddressCard,
  faArrowCircleRight,
  faBars,
  faBell,
  faBold,
  faCaretDown,
  faCaretUp,
  faCheck,
  faCheckSquare,
  faChevronDown,
  faChevronUp,
  faChevronLeft,
  faChevronRight,
  faCircle,
  faCode,
  faCoffee,
  faCog,
  faComment,
  faCommentAltLines,
  faEdit,
  fasEdit,
  faEnvelope,
  faExclamationCircle,
  faExclamationTriangle,
  faExpandArrowsAlt,
  faExternalLinkSquare,
  faEye,
  faFacebook,
  faFacebookF,
  faFacebookSquare,
  faInfoCircle,
  faItalic,
  faLink,
  faListOl,
  faListUl,
  faLock,
  faMinus,
  faMobileAlt,
  faParagraph,
  faPlayCircle,
  faPlus,
  faQuestionCircle,
  faQuoteRight,
  faRedo,
  farBars,
  farTimes,
  fasBars,
  fasCircle,
  fasEnvelope,
  faShare,
  faShareSquare,
  faSignIn,
  faSignOut,
  fasLock,
  faSortDown,
  faSpinner,
  faSquare,
  fasTimes,
  faUsers,
  faStrikethrough,
  faTimes,
  faTrashAlt,
  faTwitter,
  faUnderline,
  faUndo,
  faUser,
  faUserEdit,
  faUserPlus,
  farUserEdit,
)

dom.watch()

Vue.component('FontAwesomeIcon', FontAwesomeIcon)
Vue.component('FontAwesomeLayers', FontAwesomeLayers)
Vue.component('FontAwesomeLayersText', FontAwesomeLayersText)
