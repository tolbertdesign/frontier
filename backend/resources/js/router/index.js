import Vue from 'vue'
import VueRouter from 'vue-router'
import AppContainer from '@/layouts/AppContainer'
import NProgress from 'nprogress'
import ParentDashboard from '@/views/ParentDashboard'
import TeacherDashboard from '@/views/TeacherDashboard'
import EditParticipant from '@/views/EditParticipant'
import EditProfile from '@/views/EditProfile'
import EasyEmailer from '@/views/EasyEmailer/EasyEmailer'
import FinishLine from '@/views/FinishLine'
import store from '@/store'

Vue.use(VueRouter)

const routes = [
  {
    path: '/v3/home',
    component: AppContainer,
    children: [
      {
        path: 'dashboard',
        name: 'parent-dashboard',
        component: ParentDashboard,
      },
      {
        path: 'teacher-dashboard',
        name: 'teacher-dashboard',
        component: TeacherDashboard,
      },
      {
        path: 'edit-participant/:id',
        name: 'edit-participant',
        component: EditParticipant,
      },
      {
        path: 'edit-profile',
        name: 'edit-profile',
        component: EditProfile,
      },
      {
        path: 'easy-emailer/:participantUserId',
        name: 'easy-emailer',
        component: EasyEmailer,
      },
      {
        path: 'finish-line/:participantUserId',
        name: 'finish-line',
        component: FinishLine,
      },
    ],
  },

]

const router = new VueRouter({
  mode: 'history',
  forcePageRouteRefresh: false,
  linkActiveClass: 'is-active',
  routes,

  scrollBehavior (to, from, savedPosition) {
    return { x: 0, y: 0 }
  },
})

router.beforeEach((routeTo, routeFrom, next) => {
  NProgress.start()
  store.dispatch('setContentGroup', routeTo)

  if (router.options.forcePageRouteRefresh) {
    window.location.href = '/v3/home/dashboard/' + routeTo.path
    router.options.forcePageRouteRefresh = false
  } else {
    next()
  }
})

router.afterEach(() => {
  NProgress.done()
})

export default router
