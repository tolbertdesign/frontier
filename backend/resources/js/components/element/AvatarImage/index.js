import AvatarImage from './AvatarImage.vue'

export default AvatarImage
