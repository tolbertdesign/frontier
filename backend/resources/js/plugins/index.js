import 'nprogress/nprogress.css'
// import '@fortawesome/fontawesome-free/css/all.min.css'

import '@/plugins/buefy'
import '@/plugins/fontawesome'
import '@/plugins/gravatar'
import '@/plugins/tooltips'
import '@/plugins/vue-croppa'
import '@/plugins/vue-meta'
import '@/plugins/vuelidate'

// import '@/plugins/validation'
