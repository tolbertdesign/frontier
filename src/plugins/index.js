import '@/plugins/gravatar'
import '@/plugins/vue-particles'
import '@/plugins/portal-vue'
import '@/plugins/tooltips'
// import '@/plugins/validation'
