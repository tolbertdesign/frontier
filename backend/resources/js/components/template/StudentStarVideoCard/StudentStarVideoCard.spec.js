import { shallowMount, createLocalVue } from '@vue/test-utils'
import AvatarImage from '@/components/element/AvatarImage'
import router from '@/router'

import StudentStarVideoCard from './StudentStarVideoCard.vue'

import '@/plugins/fontawesome'

const localVue = createLocalVue()

describe('StudentStarVideoCard', () => {
  const $store = {
    state: {
      lang: {
        student_star_video: {},
      },
    },
    gtag: jest.fn(),
  }
  const participant = {
    profile: {
      image_name: null,
    },
    participant_info: {
      pledges: [],
    },
  }

  it('matches snapshot', () => {
    const wrapper = shallowMount(StudentStarVideoCard, {
      localVue,
      router,
      mocks: { $store },
      propsData: { participant },
      components: { AvatarImage },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
