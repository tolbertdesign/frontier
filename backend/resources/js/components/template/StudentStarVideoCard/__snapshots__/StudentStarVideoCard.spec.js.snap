// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentStarVideoCard matches snapshot 1`] = `
<div class="flex w-full">
  <div class="mr-1 sm:mr-4"><a href="/v3/tkdashboard/?redirect=auth/login/undefined/edit-personalize">
      <avatarimage-stub alt="" src="/v3-assets/dashboard/images/dashboard-avatar.svg"></avatarimage-stub>
    </a></div>
  <div class="flex-1 self-center">
    <h2>

    </h2>
  </div>
  <div class="self-center text-right">
    <routerlink-stub to="[object Object]" tag="a" event="click" class="inline-block py-1 sm:py-2 px-4 sm:px-6 text-white hover:text-white shadow font-semibold rounded-full bg-secondary whitespace-no-wrap">

    </routerlink-stub>
  </div>
</div>
`;
